 Per Peter Tribble, Setting __STDC_VERSION__ to a nonsense value won't lead to a successful outcome.
--- numpy-1.26.3/setup.py.old	2024-02-03 23:35:54.752966161 -0500
+++ numpy-1.26.3/setup.py	2024-02-03 23:36:13.220335640 -0500
@@ -257,7 +257,6 @@
             # from STL since extensions are build by C linker and
             # without C++ runtime dependencies.
             '-fno-threadsafe-statics',
-            '-D__STDC_VERSION__=0',  # for compatibility with C headers
             '-fno-exceptions',  # no exception support
             '-fno-rtti'  # no runtime type information
         ])
